src/main/java/org/ohdsi/circe/cohortdefinition/builders/CriteriaSqlBuilder.java

```diff
@@ -30,25 +30,29 @@ public String getCriteriaSql(T criteria, BuilderOptions options) {
     query = embedJoinClauses(query, joinClauses);
     query = embedWhereClauses(query, whereClauses);
 
+    query = embedAdditionalColumns(query, criteria, options);
+
+    return query;
+  }
+
+  protected String embedAdditionalColumns(String query, T criteria, BuilderOptions options) {
     if (options != null) {
       List<CriteriaColumn> filteredColumns = options.additionalColumns.stream()
-              .filter((column) -> !this.getDefaultColumns().contains(column))
-              .collect(Collectors.toList());
+        .filter((column) -> !this.getDefaultColumns().contains(column))
+        .collect(Collectors.toList());
       if (filteredColumns.size() > 0) {
-        query = StringUtils.replace(query, "@additionalColumns", ", " + this.getAdditionalColumns(filteredColumns));
+        return StringUtils.replace(query, "@additionalColumns", ", " + this.getAdditionalColumns(criteria, filteredColumns));
```

Comment:
I believe changing the 3 query assignments into 3 individual returns doesn't improve anything with the code, and makes it harder to debug when you just want to know what the query result that will be returned.